drivers/soundwire/debugfs.c

```diff
@@ -137,6 +137,60 @@ static int sdw_slave_reg_show(struct seq_file *s_file, void *data)
 }
 DEFINE_SHOW_ATTRIBUTE(sdw_slave_reg);
 
+static ssize_t sdw_slave_dev_num_read(struct file *file,
+				      char __user *to, size_t count, loff_t *ppos)
+{
+	struct sdw_slave *slave = file->private_data;
+	char buf[8] = { 0 };
+	int len;
+
+	len = scnprintf(buf, sizeof(buf), "%d\n", slave->dev_num);
+
+	return simple_read_from_buffer(to, count, ppos, buf, len);
+}
+
+static ssize_t sdw_slave_dev_num_write(struct file *file,
+				       const char __user *from, size_t count, loff_t *ppos)
+{
+	struct sdw_slave *slave = file->private_data;
+	char buf[8] = { 0 };
+	u8 dev_num = 0;
+	int ret, err;
+
+	ret = simple_write_to_buffer(buf, sizeof(buf) - 1, ppos, from, count);
+	err = kstrtou8(buf, 0, &dev_num);
+	if (err)
+		return err;
+
+	/* we can only clear the existing device number */
+	if (dev_num)
+		return -EINVAL;
+
+	/* make sure the device is active */
+	err = pm_runtime_get_sync(&slave->dev);
+	if (err < 0 && err != -EACCES) {
+		pm_runtime_put_noidle(&slave->dev);
+		return err;
+	}
+
```

Comment:
@plbossart When I tested this on ADLP RVP, I need to add some delay (like 200ms) here. Because, rt711 will take some time when it is attached after runtime resume. Otherwise, the slave->dev_num will be changed to 0 while rt711 is writing its registers.
However, slave->dev will be 0 after `echo 0 > device_number` after adding some delay here. `sdw_assign_device_num` will not be called due to slave->status is SDW_SLAVE_ATTACHED in the [if (slave->status != SDW_SLAVE_UNATTACHED)](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/bus.c#L864) test.
rt711 is already attached when `pm_runtime_get_sync` is called at line 170.